ui/src/telegrafs/components/CollectorList.tsx

```diff
@@ -83,12 +85,78 @@ export default class CollectorList extends PureComponent<Props> {
         <CollectorRow
           key={collector.id}
           collector={collector}
-          bucket={getDeep<string>(collector, 'plugins.0.config.bucket', '')}
-          onDelete={onDelete}
-          onUpdate={onUpdate}
+          onDelete={(telegraf: Telegraf) =>
+            onDeleteTelegraf(telegraf.id, telegraf.name)
+          }
+          onUpdate={onUpdateTelegraf}
           onFilterChange={onFilterChange}
         />
       ))
     }
   }
 }
+
+const mstp = (state: AppState): StateProps => ({
+  collectors: state.telegrafs.list,
+})
+
+const mdtp: DispatchProps = {
+  onUpdateTelegraf: updateTelegraf,
+  onDeleteTelegraf: deleteTelegraf,
+}
+
+export default connect<StateProps, DispatchProps, OwnProps>(
+  mstp,
+  mdtp
+)(CollectorList)
+
+type FilteredOwnProps = OwnProps & {
+  searchTerm: string
+}
+
+type FilteredProps = Props & FilteredOwnProps
+
+class FilteredCollectorList extends PureComponent<FilteredProps> {
+  render() {
+    const {
+      searchTerm,
+      collectors,
+      emptyState,
+      onFilterChange,
+      sortKey,
+      sortDirection,
+      sortType,
+      onClickColumn,
+      onUpdateTelegraf,
+      onDeleteTelegraf,
+    } = this.props
+    return (
+      <FilterList<Telegraf>
+        searchTerm={searchTerm}
+        searchKeys={['metadata.buckets[]', 'name', 'labels[].name']}
+        list={collectors}
+      >
+        {cs => (
+          <CollectorList
+            collectors={cs}
+            emptyState={emptyState}
+            onFilterChange={onFilterChange}
+            sortKey={sortKey}
+            sortDirection={sortDirection}
+            sortType={sortType}
+            onClickColumn={onClickColumn}
+            onUpdateTelegraf={onUpdateTelegraf}
+            onDeleteTelegraf={onDeleteTelegraf}
+          />
+        )}
+      </FilterList>
+    )
+  }
+}
+
+const FilteredList = connect<StateProps, DispatchProps, FilteredOwnProps>(
+  mstp,
+  mdtp
+)(FilteredCollectorList)
+
```

Comment:
i'm not sure if this would solve it, but it seems that `filteredCollectorList` is more what is wanted. what if we just exported it and stop exporting `collectorList`? again, i'm really not sure if that's what you're talking about or if it's the proper way

Reply:
>what if we just exported it and stop exporting `collectorList`

I guess I should have said: our convention around components seems to be one export per file, not one component per file. so having a helper component that isn't exposed and acts like a private member of the exported component (the exported component is in essence, this file's interface) seems reasonable

Reply:
@glinton do you mean to only export `FilteredCollectorList` or do you want to export both components?

Reply:
ya, i don't think the other component is used anywhere, so to fit convention i un-exported it